diff --git a/components/bt/controller/lib_esp32 b/components/bt/controller/lib_esp32
index 45e3280535..a4d7731a95 160000
--- a/components/bt/controller/lib_esp32
+++ b/components/bt/controller/lib_esp32
@@ -1 +1 @@
-Subproject commit 45e328053591081abcd387ec2eecc8add43f4b15
+Subproject commit a4d7731a95db8a6cfb98e5068b6757c32ecfaa2a
diff --git a/components/bt/controller/lib_esp32c2/esp32c2-bt-lib b/components/bt/controller/lib_esp32c2/esp32c2-bt-lib
index b9a902c355..3878fc3d68 160000
--- a/components/bt/controller/lib_esp32c2/esp32c2-bt-lib
+++ b/components/bt/controller/lib_esp32c2/esp32c2-bt-lib
@@ -1 +1 @@
-Subproject commit b9a902c3551ef4a2032b6662a4cbb018125ddfda
+Subproject commit 3878fc3d687ed70798952e25bdb2514e023d80dc
diff --git a/components/bt/controller/lib_esp32c3_family b/components/bt/controller/lib_esp32c3_family
index 76ed4114ee..29d5555ca1 160000
--- a/components/bt/controller/lib_esp32c3_family
+++ b/components/bt/controller/lib_esp32c3_family
@@ -1 +1 @@
-Subproject commit 76ed4114ee7d081435a3c65793b4c8eb1dfaf199
+Subproject commit 29d5555ca1febeb132f5a13556893f3419d2d640
diff --git a/components/bt/controller/lib_esp32c6/esp32c6-bt-lib b/components/bt/controller/lib_esp32c6/esp32c6-bt-lib
index 6663362597..405de68173 160000
--- a/components/bt/controller/lib_esp32c6/esp32c6-bt-lib
+++ b/components/bt/controller/lib_esp32c6/esp32c6-bt-lib
@@ -1 +1 @@
-Subproject commit 6663362597110007a901eed9c87fa1cff6ac25fe
+Subproject commit 405de68173533500319e9e2eb1c5bb13ca0a60f5
diff --git a/components/bt/controller/lib_esp32h2/esp32h2-bt-lib b/components/bt/controller/lib_esp32h2/esp32h2-bt-lib
index 9bf31906cd..586a227264 160000
--- a/components/bt/controller/lib_esp32h2/esp32h2-bt-lib
+++ b/components/bt/controller/lib_esp32h2/esp32h2-bt-lib
@@ -1 +1 @@
-Subproject commit 9bf31906cdbef6ac3d271117f1cd439bd411eef5
+Subproject commit 586a22726421eff1bf1bd6e57f378de1f51b8f72
diff --git a/components/bt/host/nimble/nimble b/components/bt/host/nimble/nimble
index fdc3526193..a1f5999ee8 160000
--- a/components/bt/host/nimble/nimble
+++ b/components/bt/host/nimble/nimble
@@ -1 +1 @@
-Subproject commit fdc352619392a6f67622d7d07403de4794051362
+Subproject commit a1f5999ee87d73c1d71e1e90991fafde4a951ec3
diff --git a/components/esp_wifi/lib b/components/esp_wifi/lib
index fff3460aaf..fda241ceff 160000
--- a/components/esp_wifi/lib
+++ b/components/esp_wifi/lib
@@ -1 +1 @@
-Subproject commit fff3460aafdcfef2d1890caa80c1855eb5fac327
+Subproject commit fda241ceff83a34609c9fc132d3e19916969f2c8
diff --git a/components/json/cJSON b/components/json/cJSON
index 87d8f0961a..acc76239be 160000
--- a/components/json/cJSON
+++ b/components/json/cJSON
@@ -1 +1 @@
-Subproject commit 87d8f0961a01bf09bef98ff89bae9fdec42181ee
+Subproject commit acc76239bee01d8e9c858ae2cab296704e52d916
diff --git a/components/lwip/lwip b/components/lwip/lwip
index f79221431f..4297782bf9 160000
--- a/components/lwip/lwip
+++ b/components/lwip/lwip
@@ -1 +1 @@
-Subproject commit f79221431fa9042b3572d271d687de66da7560c4
+Subproject commit 4297782bf9e614be1bb8605f30d46af8697fab17
diff --git a/components/openthread/lib b/components/openthread/lib
index fcee30db4b..34d698a274 160000
--- a/components/openthread/lib
+++ b/components/openthread/lib
@@ -1 +1 @@
-Subproject commit fcee30db4b7342de4df9105bdd049a09d2d63187
+Subproject commit 34d698a274940730901b934caa023a3281aca53e
diff --git a/components/openthread/openthread b/components/openthread/openthread
index 456c448284..be7d36e4ff 160000
--- a/components/openthread/openthread
+++ b/components/openthread/openthread
@@ -1 +1 @@
-Subproject commit 456c448284486abe2a9118a9fdaa468fe2383fcd
+Subproject commit be7d36e4ff9cf7df6dfce54e58a31163c87b93f7
diff --git a/components/wpa_supplicant/esp_supplicant/include/esp_dpp.h b/components/wpa_supplicant/esp_supplicant/include/esp_dpp.h
index 787717408d..e02b283c29 100644
--- a/components/wpa_supplicant/esp_supplicant/include/esp_dpp.h
+++ b/components/wpa_supplicant/esp_supplicant/include/esp_dpp.h
@@ -16,7 +16,7 @@
 extern "C" {
 #endif
 
-#define ESP_DPP_AUTH_TIMEOUT_SECS 1
+#define ESP_DPP_AUTH_TIMEOUT_SECS 2
 
 #define ESP_ERR_DPP_FAILURE         (ESP_ERR_WIFI_BASE + 151)   /*!< Generic failure during DPP Operation */
 #define ESP_ERR_DPP_TX_FAILURE      (ESP_ERR_WIFI_BASE + 152)   /*!< DPP Frame Tx failed OR not Acked */
diff --git a/components/wpa_supplicant/esp_supplicant/src/esp_dpp.c b/components/wpa_supplicant/esp_supplicant/src/esp_dpp.c
index 9279f340dc..b6037d51bf 100644
--- a/components/wpa_supplicant/esp_supplicant/src/esp_dpp.c
+++ b/components/wpa_supplicant/esp_supplicant/src/esp_dpp.c
@@ -88,6 +88,7 @@ static uint8_t esp_dpp_deinit_auth(void)
 static void esp_dpp_call_cb(esp_supp_dpp_event_t evt, void *data)
 {
     if (evt == ESP_SUPP_DPP_FAIL && s_dpp_ctx.dpp_auth) {
+        wpa_printf(MSG_DEBUG, "called esp_dpp_call_cb\n");
         esp_dpp_deinit_auth();
     }
     s_dpp_ctx.dpp_event_cb(evt, data);
@@ -95,10 +96,11 @@ static void esp_dpp_call_cb(esp_supp_dpp_event_t evt, void *data)
 
 static void esp_dpp_auth_conf_wait_timeout(void *eloop_ctx, void *timeout_ctx)
 {
+    wpa_printf(MSG_DEBUG, "called esp_dpp_auth_conf_wait_timeout\n");
     if (!s_dpp_ctx.dpp_auth || !s_dpp_ctx.dpp_auth->waiting_auth_conf) {
         return;
     }
-
+    printf("DPP: Terminate authentication exchange due to Auth Confirm timeout");
     wpa_printf(MSG_DEBUG,
                "DPP: Terminate authentication exchange due to Auth Confirm timeout");
     esp_dpp_call_cb(ESP_SUPP_DPP_FAIL, (void *)ESP_ERR_DPP_AUTH_TIMEOUT);
@@ -107,6 +109,7 @@ static void esp_dpp_auth_conf_wait_timeout(void *eloop_ctx, void *timeout_ctx)
 esp_err_t esp_dpp_send_action_frame(uint8_t *dest_mac, const uint8_t *buf, uint32_t len,
                                     uint8_t channel, uint32_t wait_time_ms)
 {
+    wpa_printf(MSG_DEBUG, "Called esp_dpp_send_action_frame\n");
     wifi_action_tx_req_t *req = os_zalloc(sizeof(*req) + len);;
     if (!req) {
         return ESP_FAIL;
@@ -119,9 +122,12 @@ esp_err_t esp_dpp_send_action_frame(uint8_t *dest_mac, const uint8_t *buf, uint3
     req->rx_cb = s_action_rx_cb;
     memcpy(req->data, buf, req->data_len);
 
-    wpa_printf(MSG_DEBUG, "DPP: Mgmt Tx - MAC:" MACSTR ", Channel-%d, WaitT-%d",
+    printf("DPP: Mgmt Tx - MAC:" MACSTR ", Channel-%d, WaitT-%d",
                MAC2STR(dest_mac), channel, wait_time_ms);
 
+    //wpa_printf(MSG_DEBUG, "DPP: Mgmt Tx - MAC:" MACSTR ", Channel-%d, WaitT-%d",
+    //           MAC2STR(dest_mac), channel, wait_time_ms);
+
     if (ESP_OK != esp_wifi_action_tx_req(WIFI_OFFCHAN_TX_REQ, channel,
                                          wait_time_ms, req)) {
         wpa_printf(MSG_ERROR, "DPP: Failed to perform offchannel operation");
@@ -258,12 +264,15 @@ static void esp_dpp_rx_auth_conf(struct action_rx_param *rx_param, uint8_t *dpp_
                MAC2STR(rx_param->sa));
 
     if (!auth) {
-        wpa_printf(MSG_DEBUG, "DPP: No DPP Authentication in progress - drop");
+        printf("DPP: No DPP Authentication in progress - drop");
+        wpa_printf(MSG_MSGDUMP, "DPP: No DPP Authentication in progress - drop");
         rc = ESP_ERR_DPP_FAILURE;
         goto fail;
     }
 
     if (os_memcmp(rx_param->sa, auth->peer_mac_addr, ETH_ALEN) != 0) {
+        printf("DPP: MAC address mismatch (expected "
+                   MACSTR ") - drop", MAC2STR(auth->peer_mac_addr));
         wpa_printf(MSG_DEBUG, "DPP: MAC address mismatch (expected "
                    MACSTR ") - drop", MAC2STR(auth->peer_mac_addr));
         rc = ESP_ERR_DPP_FAILURE;
@@ -272,6 +281,7 @@ static void esp_dpp_rx_auth_conf(struct action_rx_param *rx_param, uint8_t *dpp_
 
     eloop_cancel_timeout(esp_dpp_auth_conf_wait_timeout, NULL, NULL);
 
+    printf("esp_dpp.c: Called dpp_auth_conf_rx\n");
     if (dpp_auth_conf_rx(auth, (const u8 *)&public_action->v,
                          dpp_data, len) < 0) {
         wpa_printf(MSG_DEBUG, "DPP: Authentication failed");
@@ -663,6 +673,8 @@ static void offchan_event_handler(void *arg, esp_event_base_t event_base,
     if (event_id == WIFI_EVENT_ACTION_TX_STATUS) {
         wifi_event_action_tx_status_t *evt =
             (wifi_event_action_tx_status_t *)event_data;
+        printf("Mgmt Tx Status - %d, Cookie - 0x%x",
+                   evt->status, (uint32_t)evt->context);
         wpa_printf(MSG_DEBUG, "Mgmt Tx Status - %d, Cookie - 0x%x",
                    evt->status, (uint32_t)evt->context);
 
@@ -671,6 +683,7 @@ static void offchan_event_handler(void *arg, esp_event_base_t event_base,
             if (s_dpp_listen_in_progress) {
                 esp_supp_dpp_stop_listen();
             }
+            wpa_printf(MSG_DEBUG, "Error\n");
             esp_dpp_call_cb(ESP_SUPP_DPP_FAIL, (void *)ESP_ERR_DPP_TX_FAILURE);
         }
 
diff --git a/components/wpa_supplicant/esp_supplicant/src/esp_dpp_i.h b/components/wpa_supplicant/esp_supplicant/src/esp_dpp_i.h
index 6761e86ac2..b242c81867 100644
--- a/components/wpa_supplicant/esp_supplicant/src/esp_dpp_i.h
+++ b/components/wpa_supplicant/esp_supplicant/src/esp_dpp_i.h
@@ -35,7 +35,7 @@ typedef struct {
 } dpp_event_t;
 
 #define BOOTSTRAP_ROC_WAIT_TIME 500
-#define OFFCHAN_TX_WAIT_TIME 500
+#define OFFCHAN_TX_WAIT_TIME 2000
 
 struct dpp_bootstrap_params_t {
     enum dpp_bootstrap_type type;
diff --git a/components/wpa_supplicant/src/common/dpp.c b/components/wpa_supplicant/src/common/dpp.c
index b5d57d1e01..4f48f3b31a 100644
--- a/components/wpa_supplicant/src/common/dpp.c
+++ b/components/wpa_supplicant/src/common/dpp.c
@@ -6,6 +6,7 @@
  * This software may be distributed under the terms of the BSD license.
  * See README for more details.
  */
+#define DEBUG_PRINT 1
 
 #include "utils/includes.h"
 #include <fcntl.h>
@@ -20,7 +21,7 @@
 #include "crypto/aes_siv.h"
 #include "crypto/sha256.h"
 #include "dpp.h"
-
+#include "utils/wpa_debug.h"
 static const char * dpp_netrole_str(enum dpp_netrole netrole);
 
 #ifdef CONFIG_TESTING_OPTIONS
@@ -3414,6 +3415,8 @@ fail:
 int dpp_auth_conf_rx(struct dpp_authentication *auth, const u8 *hdr,
 		     const u8 *attr_start, size_t attr_len)
 {
+
+	wpa_printf(MSG_DEBUG, "Called dpp_auth_conf_rx\n");
 	const u8 *r_bootstrap, *i_bootstrap, *wrapped_data, *status, *i_auth;
 	u16 r_bootstrap_len, i_bootstrap_len, wrapped_data_len, status_len,
 		i_auth_len;
diff --git a/components/wpa_supplicant/src/utils/wpa_debug.c b/components/wpa_supplicant/src/utils/wpa_debug.c
index 322a52a30b..d24182fe19 100644
--- a/components/wpa_supplicant/src/utils/wpa_debug.c
+++ b/components/wpa_supplicant/src/utils/wpa_debug.c
@@ -6,6 +6,7 @@
  * See README for more details.
  */
 #ifdef ESP_SUPPLICANT
+#define DEBUG_PRINT 1
 #include "utils/includes.h"
 #include "utils/common.h"
 #include "utils/wpa_debug.h"
